lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -254,32 +262,32 @@ func (s *subscriberdbServicer) loadSubscribersPageFromCache(networkID string, re
 
 // getDigestInfo returns the correctly formatted Digest and NoUpdates values
 // according to the client digest.
-func (s *subscriberdbServicer) getDigestInfo(clientDigest *lte_protos.Digest, networkID string) (*lte_protos.Digest, bool) {
-	digest, err := storage.GetDigest(s.digestStore, networkID)
+func (s *subscriberdbServicer) getDigestInfo(clientDigest *protos.Digest, networkID string) (*protos.DigestTree, bool) {
+	digestTree, err := syncstore.GetDigestTree(s.store, networkID)
 	// If digest generation fails, the error is swallowed to not affect the main functionality
 	if err != nil {
-		glog.Errorf("Generating digest for network %s failed: %+v", networkID, err)
-		return &lte_protos.Digest{Md5Base64Digest: ""}, false
+		glog.Errorf("Load digest for network %s failed: %+v", networkID, err)
+		return &protos.DigestTree{RootDigest: &protos.Digest{Md5Base64Digest: ""}}, false
 	}
+	rootDigest := digestTree.RootDigest.GetMd5Base64Digest()
+	noUpdates := rootDigest != "" && rootDigest == clientDigest.GetMd5Base64Digest()
 
-	noUpdates := digest != "" && digest == clientDigest.GetMd5Base64Digest()
-	digestProto := &lte_protos.Digest{Md5Base64Digest: digest}
-	return digestProto, noUpdates
+	return digestTree, noUpdates
 }
 
 // shouldResync returns whether a gateway requires an Orc8r-directed resync by checking its
 // last resync time.
 func (l *subscriberdbServicer) shouldResync(network string, gateway string) bool {
-	lastResyncTime, err := l.lastResyncTimeStore.Get(network, gateway)
+	lastResyncTime, err := l.store.GetLastResync(network, gateway)
 	// If check last resync time in store fails, swallow the error and stick to the original callpath
 	if err != nil {
 		glog.Errorf("check last resync time of gateway %+v of network %+v: %+v", gateway, network, err)
 		return false
 	}
 	// Add a deterministic jitter to AGW sync intervals in the range of
 	// [0, 0.5 * resyncIntervalSecs] to ameliorate the thundering herd effect
-	resyncIntervalJitter := math.JitterUint32(l.resyncIntervalSecs, gateway, 0.5)
-	shouldResync := uint32(time.Now().Unix())-lastResyncTime > l.resyncIntervalSecs+resyncIntervalJitter
+	resyncIntervalJitter := math.JitterInt64(l.resyncIntervalSecs, gateway, 0.5)
```

Comment:
Why Int64 instead of Uint64?

Reply:
Because oftentimes `lastResyncTime` is used arithmetically with `time.Now().Unix()` (which is `int64`), so using `int64` saves more type casting @hcgatewood

Reply:
Cool, makes sense